foundry/gui/orb.py

```diff
@@ -0,0 +1,362 @@
+            state.move_timer_to_exit = action.payload
+
+        elif action.type == Actions.TOUCH_GAME_TIMER_STOPS:
+            state.touch_game_timer_stops = action.payload
+
+        return state
+
+
+class RomInterface:
+    """Handles all the read/write operations to the ROM.
+
+    This reads out the initial state of the ROM into a new State object.  It
+    can also take in a State object and write that into the ROM data.
+    """
+
+    def __init__(self, rom: Rom):
+        """Create all of the CodeEdit objects for all UI selections."""
+        self.rom = rom
+        nop_x_3 = bytearray([0xEA, 0xEA, 0xEA])
+
+        self._move_timer_to_exit = CodeEditDict(
+            rom,
+            0x68FD,
+            3,
+            bytearray([0x18, 0x05, 0xF0, 0x12]),
+            {False: bytearray([0x8c, 0xf4, 0x7c]), True: nop_x_3},
+            bytearray([0x88, 0xd0, 0x0b, 0x8c]),
+        )
+
+        self._move_touch_to_timer = CodeEditDict(
+            rom,
+            0x6913,
+            3,
+            bytearray([0xb5, 0x9a, 0xf0, 0x0e]),
+            {False: bytearray([0x8d, 0xf4, 0x7c]), True: nop_x_3},
+            bytearray([0x20, 0x12, 0xc4, 0xd0]),
+        )
+
+        self._stop_timer = self.StrayOrb(rom)
+
+    def read_state(self) -> State:
+        """Reads the ROM and creates a cooresponding abstract State instance."""
+        return State(
+            self._move_touch_to_timer.read(),
+            self._move_timer_to_exit.read(),
+            self._stop_timer.read()
+        )
+
+    def write_state(self, state: State):
+        """Takes in an abstract State instance, and writes to the ROM.
+
+        NOTE: This writes to the specified ROM which is not writing to the
+        file itself.  The caller/user is responsible for requesting the changes
+        be saved to the file.
+        """
+        self._move_touch_to_timer.write(state.move_touch_to_timer)
+        self._move_timer_to_exit.write(state.move_timer_to_exit)
+        self._stop_timer.write(state.touch_game_timer_stops)
+
+    class StrayOrb:
+        def __init__(self, rom: Rom):
+            self._rom = rom
+
+            self._jump_table = CodeEditDict(
+                rom,
+                0x6014,
+                2,
+                bytearray([0x46, 0xBF, 0xFF, 0xA9]),
+                {False: bytearray([0x5B, 0xA9]), True: bytearray([0xBF, 0xBF])},
+                bytearray([0x88, 0xd0, 0x0b, 0x8c]),
+            )
+
+            self._stop_timer = CodeEditDict(
+                rom,
+                0x7FCF,
+                6,
+                bytearray([0x07, 0x4C, 0xE7, 0xD5]),
+                {False: bytearray([0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF]), True: bytearray([0xA9, 0x01, 0x8D, 0xF3, 0x05, 0x60])},
+                []
+            )
+
+        def read(self) -> bool:
+            results_list = [self._jump_table.read(), self._stop_timer.read()]
+
+            if results_list[0] != results_list[1]:
+                return None
```

Comment:
According to the type hints, this method should only return true or false.  In this case, it is also returning None.  This should use the Optional keyword to denote that None is a possibility as well.